tests/pthread/test_pthread_lsan_leak.cpp

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <thread>
+#include <emscripten.h>
+
+void *global_ptr;
+thread_local void *tls_ptr;
+std::atomic<bool> thread_done;
+
+void f(void) {
+  void *local_ptr = malloc(42);
+  memset(local_ptr, 0, 42);
```

Comment:
I'm surprised this `memset` doesn't get optimized out at higher optimization levels.

Reply:
We do not run this test with optimization.